galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistQuery.java

```diff
@@ -0,0 +1,178 @@
+			
+			JsonObject jsonObject = new JsonParser().parse(worklistProperty).getAsJsonObject();
+			JsonArray runIds = (JsonArray) jsonObject.get("runIds");
+			
+			
+			String id = null;
+			for (JsonElement el : runIds) {
+
+				id = el.toString().replaceAll("[\"]", "");
+
+				
+				IRunResult run = null;
+				try {
+					run = getRunById(id);
+				} catch (ResultArchiveStoreException e) {
+					throw new ServletException("Error while retrieving run", e);
+				}
+				
+				
+				if (run != null) {
+					
+					try {
+						
+						String runName = run.getTestStructure().getRunName();
+						String shortName = run.getTestStructure().getTestShortName();
+						String testClass = run.getTestStructure().getTestName();
+						String result = run.getTestStructure().getResult();
+						
+						WorklistItem worklistItem = new WorklistItem(id, runName, shortName, testClass, result);
+						
+						runs.add(worklistItem);
+						
+					} catch (ResultArchiveStoreException e) {
+						throw new ServletException("Error while retriving run information", e);
+					}
+					
+				} else {
+					JsonError error = new JsonError("No run found by ID: " + id);
+					PrintWriter out = resp.getWriter();
+					resp.setContentType( "Application/json");
+			        resp.addHeader("Access-Control-Allow-Origin", "*");
+			        resp.setStatus(401);
+			        out.print(gson.toJson(error));
+			        out.close();
+			         
+			        return;
+				}
+				
+			}
+			
+		}
+		
+		
+		JsonObject worklist = new JsonObject();
+		JsonElement worklistItems = gson.toJsonTree(runs);
+		worklist.add("worklistItems", worklistItems);
+		
+		
+		String json = "";
+		 
+		 if (worklist.size() != 0) {
+			 try {
+				 json = gson.toJson(worklist);
+			 } catch (Exception e) {
+				 throw new ServletException("Error retrieving page, ", e);
+			 }
+		 }
+		
+		 try {
+	         PrintWriter out = resp.getWriter();
+	         resp.setContentType( "Application/json");
+	         resp.addHeader("Access-Control-Allow-Origin", "*");
+	         out.print(json);
+	         out.close();
+	      } catch (Exception e) {
+	         throw new ServletException("An error occurred whilst retrieving the Worklist", e);
+	      }
+		
+	}
+	
+	private IRunResult getRunById(String id) throws ResultArchiveStoreException{
+
+		IRunResult run = null;
+	       
+        for (IResultArchiveStoreDirectoryService directoryService : framework.getResultArchiveStore().getDirectoryServices()) {
+           
+           run = directoryService.getRunById(id);
+           if(run != null) {
+              break;
+           }
+           
+        }
+        
+        if(run == null) {
+           return null;
+        }
+		
+		return run;
+	}
+	
+	private String getWorklistPropertyFromDss() throws DynamicStatusStoreException, ServletException {
+	
+		
+		IDynamicStatusStoreService dss = framework.getDynamicStatusStoreService("webui");
+		
+		try {
+			DynamicStatusStoreSingleton.setDss(dss);
```

Comment:
dont need this at all

Reply:
you have got the dss from the line above so you don't need to store it anywhere else.   Also singletons are BAD,  avoid if possible